### DISCUSSION TITLE
import ES Components error

### DISCUSSION THREAD

**Discussion by AllenZhcn** [remix-run/remix#5184] [Q&A]
### What version of Remix are you using?

1.4.3

### Steps to Reproduce

1. npx create-remix  / with typescript 
2. open the code at
<img width="414" alt="Screen Shot 2022-05-15 at 10 11 11 AM" src="https://user-images.githubusercontent.com/95735164/168454276-a897a27a-e234-4fd7-b132-68af5483b447.png">
3. make One Component to replace default index  Component
<img width="966" alt="Screen Shot 2022-05-15 at 10 12 37 AM" src="https://user-images.githubusercontent.com/95735164/168454340-1cefb9c9-66d5-4146-aed0-828ef1bb442d.png">
4. exec : npm run start / got error
<img width="932" alt="component" src="https://user-images.githubusercontent.com/95735164/168455786-a5ac058b-7214-435c-9341-f18d5ce1901c.png">

5. Based on the other errors, I knew it was an es component causing the error and I found the solution.Based on the other errors, I knew it was an es component causing the error and I found the [solution](https://remix.run/docs/en/v1/pages/gotchas#importing-esm-packages)
 
<img width="979" alt="es" src="https://user-images.githubusercontent.com/95735164/168455810-4282ac64-7826-4056-9cdb-02253d33cb98.png">

6. Here's my guess. react-image-annotate is a ES module, but remix is ssr framework needs cjs. So the provided method converts es to cjs. but if cjs is used inside the react-image-annotate module. I will get a require() error

### Expected Behavior

ES Component compiles properly and displays properly

### Actual Behavior

Error [ERR_REQUIRE_ESM]: Must use import to load ES Module: /Users/xxx/Code/Remix/Remix-demo/node_modules/@babel/runtime/helpers/esm/objectSpread.js
require() of ES modules is not supported.
require() of /Users/xxx/Code/Remix/Remix-demo/node_modules/@babel/runtime/helpers/esm/objectSpread.js from /Users/xxx/Code/Remix/Remix-demo/build/index.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename objectSpread.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /Users/xxx/Code/Remix/Remix-demo/node_modules/@babel/runtime/helpers/esm/package.json.

    at new NodeError (internal/errors.js:322:7)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1102:13)
    at Module.load (internal/modules/cjs/loader.js:950:32)
    at Function.Module._load (internal/modules/cjs/loader.js:790:12)
    at Module.require (internal/modules/cjs/loader.js:974:19)
    at require (internal/modules/cjs/helpers.js:93:18)
    at Object.<anonymous> (/Users/fan.hu/Code/Remix/Remix-demo/build/index.js:84:37)
    at Module._compile (internal/modules/cjs/loader.js:1085:14)
    at Object.Module._extensions..js (internal/modules/cjs/loader.js:1114:10)
    at Module.load (internal/modules/cjs/loader.js:950:32)
npm ERR! code ELIFECYCLE

**Comment by sagar-guglani** [comment]
Facing similar error

Error [ERR_REQUIRE_ESM]: require() of ES Module
/node_modules/@babel/runtime/helpers/esm/objectSpread2.js from 
/server/middleware/renderer.js not supported.
Instead change the require of objectSpread2.js in server/middleware/renderer.js to a dynamic import() which is available in all CommonJS modules.

Did you get the solution ?
